packages/babel-plugin-proposal-object-rest-spread/test/fixtures/object-rest/variable-destructuring/input.js

```diff
@@ -17,3 +17,9 @@ let {
 } = complex;
 
 let { x4: { ...y4 } } = z;
+
+let { x5: { w5, ...y5 } } = z();
+let { x6: { w6: { a6, ...y6 } } } = z();
+let { x7: { e7, r7 }, q7: { w7: { a7, ...y7 } } } = z();
+let { x8, ...y8 } = z();
+let { x9: { w9: { a9, ...y9 } }, x10: { a10, ...y10 },  } = z();
```

Comment:
Can you add a test case with mixed arrays?

```suggestion
let { x10: [{ w10, ...z10 }] } = z();
let { x11: [{ a11, b11 }, { c11, ...d11 }] } = z();
let { x12: [, { c12, ...d12 }] } = z();
```

Reply:
Just pushed up a commit. Thanks for the recommendation, it caught an edge case with the implementation.

Reply:
Can you add a new test case?
```js
const { x: [...{ x, ...y }] } = foo();
```

A RestElement's argument can be a pattern so we have to recurse into it.

Reply:
Great point! I added the test case and confirmed the output looks as I'd expect.

Reply:
We can not return early when `properties` is falsy, instead we should invoke the recursive function on `.argument`. For example,

```
const { x: [...{ ...y }] } = z();
```
has only one binding identifier but `doesAnyChildNodeHaveMoreThanOneProperty` returns `true`.

Reply:
Hey @JLHwung - thanks for the the detailed case. I looked through https://astexplorer.net/ and believe I was able to diagnose the issue you called our with more robust test cases for array destructuring. 

Happy to make any other adjustments. Thank you for the thorough review and prompt responses!